Source/WebCore/platform/graphics/cg/GraphicsContextGLCG.cpp

```diff
@@ -531,7 +535,16 @@ void GraphicsContextGL::paintToCanvas(const GraphicsContextGLAttributes& sourceC
     }));
 
     auto imageSize = pixelBuffer->size();
-    auto image = NativeImage::create(adoptCF(CGImageCreate(imageSize.width(), imageSize.height(), 8, 32, 4 * imageSize.width(), pixelBuffer->format().colorSpace.platformColorSpace(), bitmapInfo, dataProvider.get(), 0, false, kCGRenderingIntentDefault)));
+    return adoptCF(CGImageCreate(imageSize.width(), imageSize.height(), 8, 32, 4 * imageSize.width(), pixelBuffer->format().colorSpace.platformColorSpace(), bitmapInfo, dataProvider.get(), 0, false, kCGRenderingIntentDefault));
+}
+
+void GraphicsContextGL::paintToCanvas(PlatformImagePtr&& platformImage, const IntSize& canvasSize, GraphicsContext& context)
+{
+    if (canvasSize.isEmpty())
+        return;
+
+    auto image = NativeImage::create(WTFMove(platformImage));
```

Comment:
I think you could revert the NativeImage -> PlatformImagePtr changes. E.g. if only possible, use NativeImage, a real abstraction, as opposed to typedef hack.

Reply:
Nice. This cleans up the duplication of paintToCanvas between Cairo and CG.